internal/pkg/cli/svc_init.go

```diff
@@ -219,6 +219,19 @@ func (o *initSvcOpts) Validate() error {
 
 // Ask prompts for and validates any required flags.
 func (o *initSvcOpts) Ask() error {
+	if o.name != "" && o.wkldType == "" {
+		// Best effort to validate the service name without type.
+		if err := o.validateSvc(); err != nil {
+			return err
+		}
+		shouldSkipAsking, err := o.shouldSkipAsking()
+		if err != nil {
+			return err
+		}
+		if shouldSkipAsking {
+			return nil
+		}
+	}
 	if o.wkldType != "" {
 		if err := validateSvcType(o.wkldType); err != nil {
```

Comment:
What happens if the user passes in the `--svc-type` but not the `--name`? Is that use case covered? I may be missing it, but it looks like we assume we have `o.name` in this case.

Reply:
Nevermind!